runtime/src/main/java/org/corfudb/runtime/view/stream/AddressMapStreamView.java

```diff
@@ -88,7 +88,8 @@ protected ILogData removeFromQueue(NavigableSet<Long> queue) {
                     // Validate that the data entry belongs to this stream, otherwise, skip.
                     // This verification protects from sequencer regression (tokens assigned in an older epoch
                     // that were never written to, and reassigned on a newer epoch)
-                    if (ld.containsStream(this.id) && ld.getType() == DataType.DATA) {
+                    if (ld.getType() == DataType.HOLE || (ld.containsStream(this.id) &&
```

Comment:
By the way this will break existing consumers of IStreamView (i.e. transaction pollers), because now see unexpected holes.

Reply:
They should not IMO. What entries are seen by the client is dictated by getNextEntries() and that function filters out the holes.